File: eco-core/core-backend/src/main/java/com/willfp/eco/internal/factory/EcoRunnableFactory.java
package com.willfp.eco.internal.factory;

import com.willfp.eco.core.EcoPlugin;
import com.willfp.eco.core.PluginDependent;
import com.willfp.eco.core.factory.RunnableFactory;
import com.willfp.eco.core.scheduling.RunnableTask;
import com.willfp.eco.internal.scheduling.EcoRunnableTask;
import org.jetbrains.annotations.NotNull;

import java.util.function.Consumer;

public class EcoRunnableFactory extends PluginDependent<EcoPlugin> implements RunnableFactory {
    public EcoRunnableFactory(@NotNull final EcoPlugin plugin) {
        super(plugin);
    }

    @Override
    public RunnableTask create(@NotNull final Consumer<RunnableTask> consumer) {
        return new EcoRunnableTask(this.getPlugin()) {
            @Override
            public void run() {
                consumer.accept(this);
            }
        };
    }
}

File: eco-core/core-backend/src/main/kotlin/com/willfp/eco/internal/factory/EcoRunnableFactory.kt
package com.willfp.eco.internal.factory

import com.willfp.eco.core.EcoPlugin
import com.willfp.eco.core.factory.RunnableFactory
import com.willfp.eco.core.scheduling.Scheduler
import com.willfp.eco.internal.scheduling.EcoRunnableTask
import java.util.function.Consumer

class EcoRunnableFactory(private val plugin: EcoPlugin) : RunnableFactory {
    override fun create(consumer: Consumer<Scheduler>): Scheduler {
        return object : EcoRunnableTask(plugin) {
            override fun run() {
                consumer.accept(this)
            }
        }
    }
}
